cluster/azure/pod_identity_msi/main.tf

```diff
@@ -0,0 +1,30 @@
+module "azure-provider" {
+  source = "../provider"
+}
+
+data "azurerm_resource_group" "podid" {
+  name = "${var.resource_group_name}"
+}
+
+resource "azurerm_user_assigned_identity" "podid" {
+  count               = "${var.enable_pod_identity ? 1 : 0}"
+  name                = "${var.identity_name}"
+  resource_group_name = "${data.azurerm_resource_group.podid.name}"
+  location            = "${data.azurerm_resource_group.podid.location}"
+}
+
+resource "null_resource" "wait_for_identity_propagation" {
+  count               = "${var.enable_pod_identity ? 1 : 0}"
+  provisioner "local-exec" {
+    command = "${path.module}/wait_for_identity.sh ${join("",azurerm_user_assigned_identity.podid.*.principal_id)}"
+  }
```

Comment:
All calls to bash scripts in `local-exec` fail with execute permission if you're running on WSL. Should we add something like `chmod +x ${path.module}/wait_for_identity.sh`?